AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -100,17 +113,31 @@ - (id)init {
         memset(self.renderingContext, 0, sizeof(AudioRendererContext));
 
         // Setup the AVAudioEngine along with the rendering context
-        if (![self setupAudioEngine]) {
+        if (![self setupPlayoutAudioEngine]) {
             NSLog(@"Failed to setup AVAudioEngine");
         }
 
         // The manual rendering block (called in Core Audio's VoiceProcessingIO's playout callback at real time)
-        self.renderingContext->renderBlock = (__bridge void *)(_engine.manualRenderingBlock);
+        self.renderingContext->renderBlock = (__bridge void *)(_playoutEngine.manualRenderingBlock);
+
+        // Attach music node to plaoyt engine
+        [self attachMusicNodeToEngine:_playoutEngine];
 
         // Initialize the capturing context
         self.capturingContext = malloc(sizeof(AudioCapturerContext));
         memset(self.capturingContext, 0, sizeof(AudioCapturerContext));
         self.capturingContext->bufferList = &_captureBufferList;
+
+        // Setup the AVAudioEngine along with the rendering context
+        if (![self setupRecordAudioEngine]) {
+            NSLog(@"Failed to setup AVAudioEngine");
+        }
+
+        // The manual rendering block (called in Core Audio's VoiceProcessingIO's playout callback at real time)
+        self.capturingContext->renderBlock = (__bridge void *)(_recordEngine.manualRenderingBlock);
```

Comment:
Should setting the renderBlock be part of the setup process (setupRecordAudioEngine)? I was wondering if it would be missed in case the engine is destroyed and created again later.